cmake/kokkos_enable_options.cmake

```diff
@@ -65,6 +65,9 @@ KOKKOS_ENABLE_OPTION(HIP_MULTIPLE_KERNEL_INSTANTIATIONS OFF "Whether multiple ke
 KOKKOS_ENABLE_OPTION(IMPL_DESUL_ATOMICS   ON  "Whether to use desul based atomics - option only during beta")
 KOKKOS_ENABLE_OPTION(DESUL_ATOMICS_EXTERNAL OFF "Whether to use an external desul installation")
 
+# Switched on by default
+KOKKOS_ENABLE_OPTION(IMPL_OPENMP_THREADPOOL_ONLY ON "To use native OpenMP implementation post 3.6 version")
```

Comment:
Change comment to: "Use legacy OpenMP thread-pool based loop implementation."

Reply:
Should the legacy option be `ON` by default?
I think we could  use refactored (native OpenMP) version by default and only introduce this to deal with the regression for the affected systems.